src/software/world/timestamped_possession_state.h

```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+#include <map>
+#include <optional>
+
+#include "software/time/timestamp.h"
+#include "software/world/robot_state.h"
+#include "software/world/team_types.h"
+
+/**
+ * A light structure for a robot ID with TeamSide
+ */
+struct RobotIdWithTeamSide
+{
+    unsigned int id;
+    TeamSide team_side;
+
+    bool operator==(const RobotIdWithTeamSide &other) const
+    {
+        return id == other.id && team_side == other.team_side;
+    }
+
+    bool operator<(const RobotIdWithTeamSide &other) const
```

Comment:
Is this being used anywhere currently? (Edit: oh is it required for the map?)

I don't think sorting by ID is necessarily an obvious method of sorting for the type. Depending on why this is required, I'd vouch for defining the `<` operator or a comparator right before usage to make it more clear what the ordering is in that case (we've done this before with some other structs).

Reply:
yes this is used for the map